### DISCUSSION TITLE
Sometimes AudioGridder Opens Every Plugin GUI on Project Load...

### DISCUSSION THREAD

**Discussion by AnietyDev** [apohl79/audiogridder#1221] [Help]
Sometimes Audio Gridder opens every plugin's GUI when a project is loaded. When this happens CPU load/performance goes crazy. Sometimes Audio Gridder doesn't open every plugin's GUI and performance is great. I can't figure out what causes Audio Gridder to open the plugin GUI on load so that I can stop that behavior and have decent system performance.

**Comment by apohl79** [comment]
You can disable the setting "Keep Plugin UI Open" under "User Inteface" in the AG plugin.

**Comment by AnietyDev** [comment]
I should have found that on my own. For some reason I was conflating it with the Keep Plugin Window on Top setting on the server. Dropped CPU usage by a ton and made a night and day difference. Thank you and thank you for this software!

**Comment by audiogamble** [comment]
I asked a little while back about CPU issues for high-resource-usage plugin GUI's. Would this option solve my particular issue? Will the GUI be completely destroyed and removed from the host CPU with this option selected?

**Reply by apohl79** [reply]
no unfortunately not. i have an open issue regarding this. in v1.2 a plugin gui just gets hidden, as some plugins can be unstable when the gui gets destroyed. i will add the possibility to choose between hiding and destroying plugin guis in the future.

**Comment by AnietyDev** [comment]
While it doesn't solve that problem, it seems that since it doesn't load every single plugin's GUI at startup it still significantly reduces the load/impact of having plugin GUIs loaded. I don't normally open every single plugin in a project every time I sit down to work on it so preloading them all isn't worth the performance impact for me. I could be misunderstanding what's going on, but changing this seemed to significantly reduce the load. My project today I could leave my PC in silent mode and didn't need to switch to the performance power plan.

**Reply by apohl79** [reply]
It does not actually preload the UIs. This setting would just open the UI of the active plugin when the AG plugin UI is created. Without it, you will have to click on a plugin to see its UI.

The behavior of hiding plugin UIs rather than destroying them on the server side stays. But a UI will only be created once you open it in your DAW.
